src/test/java/io/confluent/connect/elasticsearch/ElasticsearchSinkTaskTest.java

```diff
@@ -85,12 +89,22 @@ public void testPutSkipNullRecords() {
     // skip null
     SinkRecord nullRecord = record(true, true, 0);
     task.put(Collections.singletonList(nullRecord));
-    verify(client, never()).index(eq(nullRecord), any(DocWriteRequest.class), any(OffsetState.class));
+    verify(client, never()).index(eq(nullRecord), any(DocWriteRequest.class), any(AsyncOffsetState.class));
 
-    // don't skip non-null
+    // don't skip non-null, asynchronous mode
+    when(context.assignment()).thenReturn(Collections.singleton(new TopicPartition(TOPIC, 1)));
+    props.put(FLUSH_SYNCHRONOUSLY_CONFIG, "false");
```

Comment:
I would consider making this test parameterized (similar to what we did with `ElasticsearchSinkTaskIT`) to fully test both modes.

Reply:
Thanks, made it parametrized.